src/packages/mikroorm/src/introspection/generate.ts

```diff
@@ -0,0 +1,234 @@
+	for (const meta of metadata.filter((m) => !m.pivotTable)) {
+		for (const prop of meta.relations) {
+			if (!prop.name.includes('Inverse')) {
+				const targetMeta = metadata.find((m) => m.className === prop.type);
+				const newProp = {
+					name: prop.name + 'Inverse',
+					type: meta.className,
+					joinColumns: prop.fieldNames,
+					referencedTableName: meta.tableName,
+					referencedColumnNames: Utils.flatten(
+						(targetMeta?.getPrimaryProps() ?? []).map((pk) => pk.fieldNames)
+					),
+					mappedBy: prop.name,
+				} as EntityProperty;
+
+				// Add reference to the inverse entity
+				const inverseMeta = metadata.find((m) => m.className === meta.className);
+				const inverseProp = inverseMeta?.props.find((p) => p.name === newProp.mappedBy);
+				if (inverseProp) inverseProp.inversedBy = newProp.name;
+
+				if (prop.reference === ReferenceType.MANY_TO_ONE) {
+					newProp.reference = ReferenceType.ONE_TO_MANY;
+				} else if (prop.reference === ReferenceType.ONE_TO_ONE && !prop.mappedBy) {
+					newProp.reference = ReferenceType.ONE_TO_ONE;
+					newProp.nullable = true;
+				} else if (prop.reference === ReferenceType.MANY_TO_MANY && !prop.mappedBy) {
+					newProp.reference = ReferenceType.MANY_TO_MANY;
+				} else {
+					continue;
+				}
+
+				targetMeta?.addProperty(newProp);
+			}
+		}
+	}
+};
+
+const detectManyToManyRelations = (metadata: EntityMetadata[], namingStrategy: NamingStrategy) => {
+	for (const meta of metadata) {
+		if (
+			meta.compositePK && // needs to have composite PK
+			meta.primaryKeys.length === meta.relations.length && // all relations are PKs
+			meta.relations.length === 2 && // there are exactly two relation properties
+			meta.relations.length === meta.props.length && // all properties are relations
+			meta.relations.every((prop) => prop.reference === ReferenceType.MANY_TO_ONE) // all relations are m:1
+		) {
+			meta.pivotTable = true;
+			const owner = metadata.find((m) => m.className === meta.relations[0].type);
+			if (!owner) throw new Error('No Owner');
+			const name = namingStrategy.columnNameToProperty(
+				meta.tableName.replace(new RegExp('^' + owner.tableName + '_'), '')
+			);
+			owner.addProperty({
+				name,
+				reference: ReferenceType.MANY_TO_MANY,
+				pivotTable: meta.tableName,
+				type: meta.relations[1].type,
+				joinColumns: meta.relations[0].fieldNames,
+				inverseJoinColumns: meta.relations[1].fieldNames,
+			} as EntityProperty);
+		}
+	}
+};
+
+const generateIdentifiedReferences = (metadata: EntityMetadata[]): void => {
+	for (const meta of metadata.filter((m) => !m.pivotTable)) {
+		for (const prop of meta.relations) {
+			if ([ReferenceType.MANY_TO_ONE, ReferenceType.ONE_TO_ONE].includes(prop.reference)) {
+				prop.wrappedReference = true;
+			}
+		}
+	}
+};
+
+const generateSingularTypeReferences = (metadata: EntityMetadata[]): void => {
+	for (const meta of metadata.filter((m) => !m.pivotTable)) {
+		meta.className = pluralize.singular(meta.className);
```

Comment:
If this is "eggs" then it becomes "egg", but if it's "egg" it stays "egg"? Is this because we don't know if the className is plural in the db?

Reply:
correct it might be "eggs" as the tablename